config/v1/types_infrastructure.go

```diff
@@ -705,7 +705,15 @@ type AlibabaCloudResourceTag struct {
 
 // NutanixPlatformSpec holds the desired state of the Nutanix infrastructure provider.
 // This only includes fields that can be modified in the cluster.
-type NutanixPlatformSpec struct{}
+type NutanixPlatformSpec struct {
+	// pcEndpoint is the endpoint (DNS name or IP address) of the Nutanix Prism Central
+	// +required
```

Comment:
I think the correct format for this is slightly different, looking at the generated CRD the field hasn't been marked as required
```suggestion
	// +kubebuilder:validation:Required
```

Reply:
I see lots of uses of "// +required" in this repo (for example, in the "config/" folder). What's the difference of these two format?  The marker "// +required" does not generate "required" marker for the field in CRD?

Reply:
The format I've suggested will add the field to the required fields within the OpenAPI schema, the current format isn't actually recognised by the generator, so doesn't add anything to the OpenAPI schema.

So with the current tag, it doesn't do anything, it will actually make a change with the new tax 

This is semi trivial because the MachineProviderConfig isn't actually a real CRD, but we have plans to generate documentation from OpenAPI schemas so it would be good to have the correct tags

Reply:
I see, will fix these.

Reply:
Fixed.